src/wake/WakeCategoryEventHandler.cpp

```diff
@@ -45,30 +48,151 @@ namespace UKControllerPlugin {
-                    + "/" + this->mapper.MapFlightplanToCategory(tagData.flightPlan);
-
-                // 15 characters is the max for tag functions, trim the aircraft type accordingly
-                if (tagString.size() > this->maxItemSize) {
-                    const unsigned int charactersToTrim = tagString.size() - this->maxItemSize;
-                    const std::string aircraftType = tagData.flightPlan.GetAircraftType();
-                    tagString = aircraftType.substr(0, aircraftType.size() - charactersToTrim) + "/"
-                        + this->mapper.MapFlightplanToCategory(tagData.flightPlan);
-                }
-
-                this->cache[tagData.flightPlan.GetCallsign()] = tagString;
+            if (tagData.itemCode == this->tagItemIdAircraftTypeCategory) {
+                tagData.SetItemString(this->GetAircraftTypeUkCategoryTagItemData(tagData));
+            } else if (tagData.itemCode == this->tagItemIdStandaloneCategory) {
+                tagData.SetItemString(this->GetStandaloneTagItemData(tagData));
+            } else if (tagData.itemCode == this->tagItemIdRecat) {
+                tagData.SetItemString(this->GetRecatTagItemData(tagData));
+            } else if (tagData.itemCode == this->tagItemIdUkRecatCombined) {
+                tagData.SetItemString(this->GetUkRecatCombinedTagItemData(tagData));
+            } else if (tagData.itemCode == this->tagItemIdAircraftTypeRecat) {
+                tagData.SetItemString(this->GetAircraftTypeRecatCategoryTagItemData(tagData));
+            }
+        }
+
+        /*
+         * Get the data for the combined Aircraft Type / UK Category item.
+         */
+        std::string WakeCategoryEventHandler::GetAircraftTypeUkCategoryTagItemData(TagData& tagData)
+        {
+            CacheItem& cached = this->FirstOrNewCacheItem(tagData.flightPlan.GetCallsign());
+            if (cached.aircraftTypeUKCategoryItem == cached.noData)
+            {
+                cached.aircraftTypeUKCategoryItem = this->GetAircraftTypeCategoryString(
+                    tagData.flightPlan.GetAircraftType(),
+                    this->ukMapper,
+                    tagData.flightPlan.GetIcaoWakeCategory()
+                );
+            }
+
+            return cached.aircraftTypeUKCategoryItem;
+        }
+
+        /*
+         * Get the data for the combined Aircraft Type / RECAT-EU category item.
+         */
+        std::string WakeCategoryEventHandler::GetAircraftTypeRecatCategoryTagItemData(TagData& tagData)
+        {
+            CacheItem& cached = this->FirstOrNewCacheItem(tagData.flightPlan.GetCallsign());
+            if (cached.aircraftTypeRecatCategoryItem == cached.noData)
+            {
+                cached.aircraftTypeRecatCategoryItem = this->GetAircraftTypeCategoryString(
+                    tagData.flightPlan.GetAircraftType(),
+                    this->recatMapper,
+                    "?"

```

Comment:
What does the question mark represent, me-lud?

Reply:
Just that we don't know the RECAT category :) It's a nice short thing rather than "None"